File: frontend-service-provider-management-system/src/app/dto/service-dto.ts
export class ServiceDTO {
    id: number;
    serviceDescription: string;
    serviceProviders: string[];
  
    constructor(id: number, serviceDescription: string, serviceProviders: string[]) {
      this.id = id;
      this.serviceDescription = serviceDescription;
      this.serviceProviders = serviceProviders;
    }
  }
  

File: backend-service-provider-management-system/src/main/java/com/comping/serviceprovidermanagementsystem/dto/ServiceDTO.java
package com.comping.serviceprovidermanagementsystem.dto;

import jakarta.validation.constraints.NotEmpty;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ServiceDTO {

    private Long id;

    @NotEmpty(message = "serviceDescription field should not be null or empty")
    private String serviceDescription;

    @NotEmpty(message = "serviceProviders field should not be null or empty")
    private List<String> serviceProviders;

}
